controllers/machinehealthcheck_targets.go

```diff
@@ -256,3 +273,76 @@ func minDuration(durations []time.Duration) time.Duration {
 	}
 	return minDuration
 }
+
+// remediate deletes the Machine given it is owned by a MachineSet and is not part of the cluster's control plane
+func (t *healthCheckTarget) remediate(logger logr.Logger, c client.Client, r record.EventRecorder) error {
+	logger = logger.WithValues("target", t.string())
+	logger.Info("Starting remediation for target")
+
+	// If the machine is not owned by a MachineSet, it should be skipped
```

Comment:
At some point, we need to decide if MHC should handle e.g. KubeadmControlPlane-owned Machines. Maybe it should handle all Machines that have a controller owner?

Reply:
@detiber I know you & I discussed KCP and remediation previously, although I don't recall where we left things. WDYT?

Reply:
I think it's probably safer to avoid auto remediation on KCP for now, either that or put it behind a feature gate.

As @JoelSpeed mentioned, we would need to likely beef up things with respect to removing etcd members that no longer exist if their backing Machine is removed without a scale down operation.